net/interfaces/interfaces.go

```diff
@@ -79,6 +79,14 @@ func isProblematicInterface(nif *net.Interface) bool {
 	if strings.HasPrefix(name, "zt") || (runtime.GOOS == "windows" && strings.Contains(name, "ZeroTier")) {
 		return true
 	}
+	// Don't try to send disco/etc packets over Cilium or Calico; they effectively
+	// DoS each other by doing traffic amplification, both of them
+	// preferring/trying to use each other for transport. See:
+	// https://github.com/tailscale/tailscale/issues/7594
+	// https://github.com/tailscale/tailscale/issues/1552
+	if strings.HasPrefix(name, "cilium") || strings.HasPrefix(name, "cali") {
```

Comment:
Do note that if you truly want to exclude all the network interfaces created by a Kubernetes CNI, the list is going to be huge.

For instance, for flannel it's `flannel.X`.

I feel like there should be a better way than excluding predefined interface names.

Reply:
Maybe a config option to prevent this behavior would be better.
Like the one discussed here: https://github.com/tailscale/tailscale/issues/1552